paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/SparkPushDownTest.scala

```diff
@@ -0,0 +1,105 @@
+    spark.sql(s"""
+                 |CREATE TABLE T (id INT, name STRING, pt STRING)
+                 |TBLPROPERTIES ('primary-key'='id, pt', 'bucket'='2')
+                 |""".stripMargin)
+
+    spark.sql("INSERT INTO T VALUES (1, 'a', 'p1'), (2, 'b', 'p1'), (3, 'c', 'p2')")
+
+    assertThat(spark.sql("SELECT * FROM T WHERE pt = 'p1'").queryExecution.optimizedPlan.exists {
+      case Filter(c: Expression, _) =>
+        c.exists {
+          case EqualTo(a: AttributeReference, l: Literal) =>
+            a.name.equals("pt") && l.value.toString.equals("p1")
+          case _ => false
+        }
+      case _ => false
+    }).isTrue
+  }
+
+  test(s"Paimon push down: apply partition filter push down with partitioned table") {
+    spark.sql(s"""
+                 |CREATE TABLE T (id INT, name STRING, pt STRING)
+                 |TBLPROPERTIES ('primary-key'='id, pt', 'bucket'='2')
+                 |PARTITIONED BY (pt)
+                 |""".stripMargin)
+
+    spark.sql("INSERT INTO T VALUES (1, 'a', 'p1'), (2, 'b', 'p1'), (3, 'c', 'p2'), (4, 'd', 'p3')")
+
+    // partition filter push down did not hit
+    assertThat(spark.sql("SELECT * FROM T WHERE id = '1'").queryExecution.optimizedPlan.exists {
+      case Filter(_: Expression, _) => true
+      case _ => false
+    }).isTrue
+    checkAnswer(spark.sql("SELECT * FROM T WHERE id = '1'"), Row(1, "a", "p1") :: Nil)
+
+    assertThat(
+      spark.sql("SELECT * FROM T WHERE id = '1' or pt = 'p1'").queryExecution.optimizedPlan.exists {
+        case Filter(_: Or, _) => true
+        case _ => false
+      }).isTrue
+    checkAnswer(
+      spark.sql("SELECT * FROM T WHERE id = '1' or pt = 'p1'"),
+      Row(1, "a", "p1") :: Row(2, "b", "p1") :: Nil)
+
+    // partition filter push down hit
+    assertThat(spark.sql("SELECT * FROM T WHERE pt = 'p1'").queryExecution.optimizedPlan.exists {
+      case Filter(_: Expression, _) => true
+      case _ => false
+    }).isFalse
+    checkAnswer(
+      spark.sql("SELECT * FROM T WHERE pt = 'p1'"),
+      Row(1, "a", "p1") :: Row(2, "b", "p1") :: Nil)
+
+    assertThat(
+      spark
+        .sql("SELECT * FROM T WHERE id = '1' and pt = 'p1'")
+        .queryExecution
+        .optimizedPlan
+        .exists {
+          case Filter(c: Expression, _) =>
+            c.exists {
+              case EqualTo(a: AttributeReference, l: Literal) =>
+                a.name.equals("pt") && l.value.toString.equals("p1")
+              case _ => false
+            }
+          case _ => false
+        }).isFalse
+    checkAnswer(spark.sql("SELECT * FROM T WHERE id = '1' and pt = 'p1'"), Row(1, "a", "p1") :: Nil)
+
+    assertThat(spark.sql("SELECT * FROM T WHERE pt < 'p3'").queryExecution.optimizedPlan.exists {
+      case Filter(_: Expression, _) => true
+      case _ => false
+    }).isFalse
+    checkAnswer(
+      spark.sql("SELECT * FROM T WHERE pt < 'p3'"),
+      Row(1, "a", "p1") :: Row(2, "b", "p1") :: Row(3, "c", "p2") :: Nil)
+  }
+
```

Comment:
maybe define two private methods (`checkFilterExists` and `checkFilterEquals`). These can be reused.